CARDS-2013 - The wrong number of steps is displayed in paginated forms when the pagination variant is `dots` or `text`

marta- opened the pull request:
See details in the description of [CARDS-2013](https://phenotips.atlassian.net/browse/CARDS-2013) on Jira.

[CARDS-2013]: https://phenotips.atlassian.net/browse/CARDS-2013?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

marta- committed:
CARDS-2013 - The wrong number of steps is displayed in paginated forms when the pagination variant is `dots` or `text`

marta- committed:
CARDS-2013 - The wrong number of steps is displayed in paginated forms when the pagination variant is `dots` or `text`
Also fixed the activeStep, which overflows for `variant` other than `progress` when the final page is saved

sashaandjic reviewed (approved):
tested

sdumitriu reviewed (changes requested):
This isn't good enough. The "current" page still goes to N+1, when the last page is saved (not in patient portal mode, but in the classical UI). Create a questionnaire with two pages, and filling it in goes:

- in progress mode: half highlighted, half done and half highlighted, all done
- in dots mode: first dot highlighted, second dot highlighted, no dot highlighted 
- in text mode: 1/2, 2/2, 3/2

The number of the "current" page after saving should also depend on the pagination variant.

marta- commented:
> This isn't good enough. The "current" page still goes to N+1, when the last page is saved (not in patient portal mode, but in the classical UI). Create a questionnaire with two pages, and filling it in goes:
> 
> * in progress mode: half highlighted, half done and half highlighted, all done
> * in dots mode: first dot highlighted, second dot highlighted, no dot highlighted
> * in text mode: 1/2, 2/2, 3/2
> 
> The number of the "current" page after saving should also depend on the pagination variant.

Addressed in 293bd97. Also rebased on the latest `dev`.

sashaandjic reviewed (approved):
tested
